### DISCUSSION TITLE
[BUG]: Clean version of package, Remove IMA SDK

### DISCUSSION THREAD

**Discussion by Rambos** [TheWidlarzGroup/react-native-video#4977] [Ideas]
### What problem are you trying to solve?

I want to remove all IMA ads related information, including the word "ads" code. I used `useExoplayerIMA` false, but there is still the word "ads" code in the project, which will cause my app to be rejected in the app market.

How can I do this?

### Proposed solution

Provide a clean version of the package with all IMA SDK / ads-related code removed (including any occurrence of the word "ads" in the code), so the app is not rejected in the app market.

### Alternatives considered

Currently downgrading to version 5.x.

Also tried setting `useExoplayerIMA` false, but the "ads" code still remains in the project.

### Platforms

iOS, Android

### Version line

v6

### Additional context

**System Version:** 33

**Architecture:** Old architecture

**On what device:** Real device

**Reproduction:** repository link

**Comment by freeboub** [comment]
Can you share the store error please ? Because there is no solution yet ...

**Comment by Rambos** [comment]
In fact, no error was reported. Some testing tools in the application market automatically scanned the related words com.google.ads, thinking that adMob sdk was introduced, and a false positive occurred.
I read the source code and found that the package name of `com.google.ads.interactivemedia.v3.api` does exist. Even though I turned on `useExoplayerIMA false`, I could still scan com.google.ads.

**Comment by freeboub** [comment]
@Rambos so play store requests you to register ads stuff ?
In fact we implemented some stub to easily disable ads that's why these classes are available in your apk ...
I think you should contact play store team to inform them it is a false positive while we try to remove it ...

[a comment by an automated account is not shown]

**Comment by freeboub** [comment]
@Rambos Did you manage to push your app ?

**Comment by ouabing** [comment]
After upgrading from v5 to v6, our app is also rejected by some markets, because they scanned and found keywords "com.google.ads" in our app.

**Comment by TWG-AI** [comment]
v7 has no IMA/ads SDK in the core library — the player is ads-free by default. Upgrading to v7 removes the IMA dependency entirely. If you need ad support in v7, a dedicated plugin path is available separately.

> I'm an AI — verify the current state of ads/IMA in v7 via the official docs.
